projects/components/src/table/data/table-cdk-data-source.ts

```diff
@@ -213,6 +216,7 @@ export class TableCdkDataSource implements DataSource<TableRow> {
     columnConfigs: TableColumnConfigExtended[],
     pageEvent: PageEvent,
     filters: TableFilter[],
+    queryProperties: Dictionary<unknown>,
```

Comment:
Similar to the comment about metadata way back - the query properties aren't being read here, we're just piping them through the table infrastructure for no clear reason. The table widget that holds them can append them to any requests for data, right?

Reply:
Combined response from [this comment](https://github.com/hypertrace/hypertrace-ui/pull/576/files#r574155027).

So, the `metaValue`. What that is, is a json configurable way to provide the information that needs to be applied to the query in a generic way when a sinlge option is applied. The shape of `metaValue` is based on the the `type` of the `TableControlOption` There are three types right now.

```typescript
export const enum TableControlOptionType {
  Filter = 'filter',
  Property = 'property',
  UnsetFilter = 'unset-filter'
}
```

1. `Filter` expects `TableFIlter`
2. `Property` expects a `Dictionary<unknown>` of query properties and their associated values
3. `UnsetFilter` expects a `string`. It uses that string to find a currently applied filter and remove it

As for the second part about table carrying all of this through. This has come up before. Its not optimal and table is the only component that really takes this approach due to the history of building it with the `TableCdkDataSource` following Material guides. But, also, the way things are currently set up, its actually necessary for `filters' due to them being manipulated by table component internally in the column header filtering. 

That's not true for `queryProperties` right now (hopefully never is). Table component doesn't manipulate them in any way so we could pull those out and not pipe them through. I really don't want to do that though because then it splits `queryProperties` from `filters` and they are really very close in how we use them. I prefer to keep them being used the same way for consistency than split them up. Besides, the `filters` were so lonely before. Now they have a friend. Do you really want to do that to them? DO YOU!?!?

Reply:
my god. I never thought about it like this (I guess we'll have to fix them both together 😉 )